pkg/providers/vsphere/vsphere.go

```diff
@@ -663,13 +672,21 @@ func (vs *VsphereTemplateBuilder) GenerateCAPISpecWorkers(clusterSpec *cluster.S
 	workerSpecs := make([][]byte, 0, len(clusterSpec.Spec.WorkerNodeGroupConfigurations))
 	for _, workerNodeGroupConfiguration := range clusterSpec.Spec.WorkerNodeGroupConfigurations {
 		values := buildTemplateMapMD(clusterSpec, *vs.datacenterSpec, vs.workerNodeGroupMachineSpecs[workerNodeGroupConfiguration.MachineGroupRef.Name], workerNodeGroupConfiguration)
-		_, ok := templateNames[workerNodeGroupConfiguration.Name]
-		if templateNames != nil && ok {
-			values["workloadTemplateName"] = templateNames[workerNodeGroupConfiguration.Name]
+
+		_, ok := workloadTemplateNames[workerNodeGroupConfiguration.Name]
+		if ok {
```

Comment:
a bit confused here (might be im not understanding it all): isn't we already calculating `workloadTemplateNames` map before calling this method? what is the case that `workerNodeGroupConfiguration.Name` key does not exist in the map?

Reply:
`GenerateCAPISpecWorkers` is called directly during `create`, and is passed `nil` kubeamconfigTemplateNames and `nil` workernodegroupconfiguration names; it then generates default values for them if they don't exist. This could be simplified

Reply:
https://github.com/aws/eks-anywhere/pull/1175/commits/41432003fb13dec72278a6d322c47c9051c4c974

Reply:
remove redundant defaulting behavior and adjust the create flow to set names